include/service/entities/plugin_service.js

```diff
@@ -1097,14 +1116,9 @@ module.exports = function PluginServiceModule(pb) {
             return cb(new Error('PluginService:[INIT] The plugin object must be passed in order to initialize the plugin'), null);
         }
 
-        var service = new pb.PluginInitializationService({
-            pluginService: this.site === plugin.site ? this : new PluginService({ site: plugin.site }),
-            pluginCache: PLUGIN_INIT_CACHE
-        });
-        service.initialize(plugin, {}, function(err, result) {
-            PLUGIN_INIT_CACHE = {};
-            cb(err, result);
-        });
+        return new pb.SitePluginInitializationService((PLUGIN_SPECS[plugin.uid]), plugin.site).initialize()
+            .then(result => cb(null, result))
```

Comment:
calling .catch on a then handler that invokes a callback will catch errors thrown in the invocation of the callback stack. Either pass both handlers into a .then invocation, e.g. `.then(result => cb(null, results), cb`, or even better use Bluebird's built in utility function, `.asCallback(cb)`

Reply:
http://bluebirdjs.com/docs/api/ascallback.html